Gems/ScriptEvents/Code/Include/ScriptEvents/ScriptEventMethod.h

```diff
@@ -111,142 +91,65 @@ namespace ScriptEvents
-                    ->Property("ReturnType", BehaviorValueProperty(&Method::m_returnType))
-                    ->Property("Parameters", BehaviorValueProperty(&Method::m_parameters))
-                    ;
-            }
-        }
-
-        AZStd::string GetName() const { return m_name.Get<AZStd::string>() ? *m_name.Get<AZStd::string>() : ""; }
-        AZStd::string GetTooltip() const { return m_tooltip.Get<AZStd::string>() ? *m_tooltip.Get<AZStd::string>() : ""; }
-        const AZ::Uuid GetReturnType() const { return m_returnType.Get<AZ::Uuid>() ? *m_returnType.Get<AZ::Uuid>() : AZ::Uuid::CreateNull(); }
-        const AZStd::vector<Parameter>& GetParameters() const { return m_parameters; }
-
-        ScriptEventData::VersionedProperty& GetNameProperty() { return m_name; }
-        ScriptEventData::VersionedProperty& GetTooltipProperty() { return m_tooltip; }
-        ScriptEventData::VersionedProperty& GetReturnTypeProperty() { return m_returnType; }
-
-        const ScriptEventData::VersionedProperty& GetNameProperty() const { return m_name; }
-        const ScriptEventData::VersionedProperty& GetTooltipProperty() const { return m_tooltip; }
-        const ScriptEventData::VersionedProperty& GetReturnTypeProperty() const { return m_returnType; }
-
-        AZ::Crc32 GetEventId() const { return AZ::Crc32(GetNameProperty().GetId().ToString<AZStd::string>().c_str()); }
+        static void Reflect(AZ::ReflectContext* context);
 
-        //! Validates that the asset data being stored is valid and supported.
-        AZ::Outcome<bool, AZStd::string> Validate() const
+        AZStd::string GetName() const
         {
-            const AZStd::string name = GetName();
-            const AZ::Uuid returnType = GetReturnType();
-
-            // Validate address type
-            if (!Types::IsValidReturnType(returnType))
+            return m_name.Get<AZStd::string>() ? *m_name.Get<AZStd::string>() : "";
+        }
+        AZStd::string GetTooltip() const
             {
-                return AZ::Failure(AZStd::string::format("The specified type %s is not valid as return type for Script Event: %s", returnType.ToString<AZStd::string>().c_str(), name.c_str()));
+            return m_tooltip.Get<AZStd::string>() ? *m_tooltip.Get<AZStd::string>() : "";
             }
 
-            // Definition name cannot be empty
-            if (name.empty())
+        const AZ::Uuid GetReturnType() const
             {
-                return AZ::Failure(AZStd::string("Definition name cannot be empty"));
+            return m_returnType.Get<AZ::Uuid>() ? *m_returnType.Get<AZ::Uuid>() : AZ::Uuid::CreateNull();
             }
 
-            // Name cannot start with a number
-            if (isdigit(name.at(0)))
+        const AZStd::vector<Parameter>& GetParameters() const
             {
-                return AZ::Failure(AZStd::string::format("%s, names cannot start with a number", name.c_str()));
+            return m_parameters;
             }
```

Comment:
It's hard to see with the diff windows, but I think the tabs might be wrong for a few functions in this file.